plugins/destination/s3/client/write.go

```diff
@@ -23,11 +23,19 @@ func (c *Client) WriteTableBatch(ctx context.Context, table *schema.Table, data
 	w := io.Writer(&b)
 	switch c.pluginSpec.Format {
 	case FormatTypeCSV:
-		if err := csv.WriteTableBatch(w, table, data); err != nil {
+		client, err := csv.NewClient()
```

Comment:
Shouldn't options be passed the same way as in read.go?
```suggestion
		opts := []csv.Options{
			csv.WithDelimiter(c.pluginSpec.Delimiter),
		}
		if c.pluginSpec.IncludeHeaders {
			opts = append(opts, csv.WithHeader())
		}

		client, err := csv.NewClient(opts...)
```